### ISSUE TITLE
Add Last Look "Server" class to `@airswap/protocols`

### ISSUE THREAD

**Issue by dmosites** [airswap/airswap-protocols#659]

**Comment by gpxl-dev** [comment]
High level implementation thoughts

Modify existing `Server` class:

- Add private field for `supportedProtocols` (array containing protocol info from `initialize`)
- Remove client initalisation from constructor into an abstract private method named `_initClient` which is only called in the constructor. Subclasses will implement this to instantiate the client, and it should set the private `client` field that already exists.
- Rename `generateRequest` to `callMethod`, and make it abstract.
- Leave all RFQ methods implemented, but short circuit rejection if `this.supportedProtocols` doesn't include RFQ.
- Create two `Server` subclasses, `WebSocketServer` and `HTTPServer`.
  - `HTTPServer` should not need to implement any _additional_ methods, as it can only be RFQ, but will need to implement `initClient`, and set `supportedProtocols` to a hard-coded RFQ Value (HTTP servers don't report their protocol support). Current logic from `generateRequest` should be moved into the `callMethod` implementation.
  - `WebSocketServer` will use [`rpc-websockets`](https://github.com/elpheria/rpc-websockets) and implement everything, plus the [Last Look methods](https://github.com/elpheria/rpc-websockets)
    - Calling a last look method on a websocket server that only supports RFQ will reject with a meaningful error
- Add a static method to named `for`, which accepts a `locator` string and `swapContract`, and returns either a `WebSocketServer` or `HTTPServer` instance.

Usage would then change as follows: 

```diff
- const server = new Server(locator, swapContract)
+ const server = Server.for(locator, swapContract)
```

**Comment by gpxl-dev** [comment]
Closed by #672
